src/auth/auth.controller.ts

```diff
@@ -196,6 +202,38 @@ export class AuthController {
         res.status(HttpStatus.OK).send({ message: "Refresh Success" });
     }
 
+    @ApiOperation({
+        summary: "Revokes user's refresh token, with a valid user id",
+        description: "using the user's id, removes user's refresh token",
+    })
+    @ApiResponse({
+        status: HttpStatus.OK,
+        description: "Refresh token successfully revoked",
+        type: GenericSuccessResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.NOT_FOUND,
+        description: "User not found.",
+        type: NotFoundErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.BAD_REQUEST,
+        description: "userId and email is provided",
+        type: BadRequestErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.FORBIDDEN,
+        description: "user doesn't have permission to preform operation",
+        type: ForbiddenErrorResponse,
+    })
+    @HttpCode(HttpStatus.OK)
+    @Roles(AppRoles.Admin)
+    @UseGuards(RolesGuard)
```

Comment:
@useGuards(RolesGuard) shouldn't be needed, as we have a app level rolesGuard 
(`src/app.module.ts`)